dataplane/src/args.rs

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+pub(crate) use clap::Parser;
+#[derive(Parser)]
+#[command(name = "Hedgehog Fabric Gateway dataplane")]
+#[command(version = "1.0")] // FIXME
+#[command(about = "A next-gen dataplane for next-gen fabric gateway", long_about = None)]
+pub(crate) struct CmdArgs {
+    #[arg(long, value_name = "core-id used as main", default_value_t = 2)]
+    main_lcore: u8,
+    #[arg(long, value_name = "map lcore set to cpu set")]
+    lcores: Option<String>,
+    #[arg(long, value_name = "in-memory flag", default_value_t = true)]
+    in_memory: bool,
+    #[arg(long, value_name = "PCI devices to probe")]
+    allow: Vec<String>,
+    #[arg(long, value_name = "huge pages", default_value_t = 8192)]
+    huge_worker_stack: u32,
+    #[arg(long, value_name = "socket memory")]
+    socket_mem: Option<String>,
+    #[arg(long, value_name = "enable/disable telemetry", default_value_t = true)]
+    no_telemetry: bool,
+    #[arg(long, value_name = "iova mode(va|pa)")]
+    iova_mode: Option<String>,
+    #[arg(long, value_name = "loglevel for a specific component")]
+    log_level: Vec<String>,
+    // other non-EAL params (NAT, routing, etc.)
+}
+impl CmdArgs {
+    pub fn eal_params(&self) -> Vec<String> {
+        let mut out = Vec::new();
+
+        out.push("--main-lcore".to_owned());
+        out.push(self.main_lcore.to_string());
+
+        out.push("--lcores".to_string());
+        out.push(
+            self.lcores
+                .clone()
+                .map_or_else(|| "2-4".to_owned(), |lcores| lcores.to_owned()),
+        );
+
+        if self.in_memory {
+            out.push("--in-memory".to_string());
+        }
+        if self.no_telemetry == false {
+            out.push("--no-telemetry".to_string());
+        } else {
+            out.push("--telemetry".to_string()); /* DPDK default */
+        }
+
+        /* IOVA mode */
+        out.push(format!(
+            "--iova-mode={}",
+            &self
+                .iova_mode
+                .clone()
+                .map_or_else(|| { "va".to_owned() }, |mode| mode.to_owned())
+        ));
+
+        /* worker huge page stack size */
+        out.push(format!("--huge-worker-stack={}", self.huge_worker_stack));
+
+        /* --allow */
+        for a in self.allow.iter() {
+            out.push("--allow".to_string());
+            out.push(a.to_owned());
+        }
+
+        // To be removed
+        if self.allow.len() == 0 {
+            out.push("--allow".to_string());
+            out.push("0000:01:00.0,dv_flow_en=1".to_string());
```

Comment:
This is a good default for the moment, but we need to be careful here.  `0000:01:00.0` is the PCI address of the CX7 card in the dev machine.  We can't count on it in general and we should let our deployment process depend on it.

That said, this is still a step in the right direction since the parameter is, at minimum, configurable now.